src/components/ui/button.tsx

```diff
@@ -42,13 +48,33 @@ const buttonVariants = cva(
 
 export interface ButtonProps
   extends React.ButtonHTMLAttributes<HTMLButtonElement>,
-    VariantProps<typeof buttonVariants> {
+    VariantProps<typeof buttonVariants>,
+    AuthorizedElementProps {
   asChild?: boolean;
 }
 
 const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
-  ({ className, variant, size, asChild = false, ...props }, ref) => {
+  (
+    { className, variant, size, asChild = false, authorizeFor, ...props },
+    ref,
+  ) => {
     const Comp = asChild ? Slot : "button";
+
+    if (authorizeFor) {
+      return (
+        <AuthorizedChild authorizeFor={authorizeFor}>
+          {({ isAuthorized }) => (
+            <Comp
+              className={cn(buttonVariants({ variant, size, className }))}
+              ref={ref}
+              disabled={!isAuthorized || props.disabled}
+              {...props}
+            />
+          )}
+        </AuthorizedChild>
+      );
+    }
+
     return (
```

Comment:
Let's avoid modifying this Button implementation.

Instead create a new wrapper like done here:
https://github.com/ohcnetwork/care_fe/pull/9296/files#diff-c29ec4c06502c45a9ee8740156618a5823f7cadbcd01a339304a66537c0dc30d

This is so that if the Button is used outside the authenticated places, it shouldn't throw an error